### PR TITLE
mirror ansible collection

### PR DESCRIPTION
**Pull request by fabricio-aguiar**
closes #5167

### CODE REVIEW

File: pulp_ansible/app/tasks/collections.py [hunk 1]
```diff
@@ -106,12 +107,15 @@ def nowhere(*args, **kwargs):
...
 
-                            collection_version_pks.append(collection_version)
+                            collection_version_pks.append(collection_version.pk)
                         import_pb.increment()
 
         collection_versions = CollectionVersion.objects.filter(pk__in=collection_version_pks)
         new_version.add_content(collection_versions)
 
+        if mirror:
+            new_version.remove_content(new_version.content.exclude(pk__in=collection_version_pks))
```
Comment (bmbouter) [inline pulp_ansible/app/tasks/collections.py]: :+1:

File: pulp_ansible/app/viewsets.py [hunk 2]
```diff
@@ -167,10 +167,15 @@ def sync(self, request, pk):
         serializer = RepositorySyncURLSerializer(data=request.data, context={"request": request})
         serializer.is_valid(raise_exception=True)
         repository = serializer.validated_data.get("repository")
+        mirror = serializer.validated_data.get("mirror", False)
```
Comment (bmbouter) [inline pulp_ansible/app/viewsets.py]: With this specified here can we de-duplicate the default by making the signature of the sync task:

`def sync(remote_pk, repository_pk, mirror):`

File: pulp_ansible/app/tasks/collections.py [hunk 3]
```diff
@@ -23,13 +23,14 @@
 log = logging.getLogger(__name__)
 
 
-def sync(remote_pk, repository_pk):
+def sync(remote_pk, repository_pk, mirror=False):
```
Comment (bmbouter) [inline pulp_ansible/app/tasks/collections.py]: the signature change would occur here

File: pulp_ansible/tests/functional/api/collection/test_sync.py [hunk 4]
```diff
@@ -82,3 +92,51 @@ def test_successive_syncs_repo_version(self):
...
+        # Step 3
+        sync(self.cfg, role_remote, repo)
+        repo = self.client.get(repo["_href"])
+        self.assertIsNotNone(repo["_latest_version_href"])
+        self.assertDictEqual(get_added_content_summary(repo), ANSIBLE_FIXTURE_CONTENT_SUMMARY)
+
+        # Step 4
+        sync(self.cfg, collection_remote, repo, mirror=True)
+        repo = self.client.get(repo["_href"])
+        self.assertIsNotNone(repo["_latest_version_href"])
```
Comment (kersommoura) [inline pulp_ansible/tests/functional/api/collection/test_sync.py]:
```suggestion

```

File: pulp_ansible/tests/functional/api/collection/test_sync.py [hunk 5]
```diff
@@ -82,3 +92,51 @@ def test_successive_syncs_repo_version(self):
...
+            ANSIBLE_COLLECTION_REMOTE_PATH,
+            gen_ansible_remote(url=ANSIBLE_GALAXY_COLLECTION_URL, whitelist=COLLECTION_WHITELIST),
+        )
+        self.addCleanup(self.client.delete, collection_remote["_href"])
+
+        # Step 3
+        sync(self.cfg, role_remote, repo)
+        repo = self.client.get(repo["_href"])
+        self.assertIsNotNone(repo["_latest_version_href"])
+        self.assertDictEqual(get_added_content_summary(repo), ANSIBLE_FIXTURE_CONTENT_SUMMARY)
```
Comment (kersommoura) [inline pulp_ansible/tests/functional/api/collection/test_sync.py]: assertEqual also verifies type. 

https://docs.python.org/3/library/unittest.html#unittest.TestCase.assertEqual

No need to update. FYI.

**Comment by bmbouter** [comment]
Thanks @fabricio-aguiar !

**Pull request merged**